specification/batch/Azure.Batch/models.tsp

```diff
@@ -0,0 +1,4083 @@
+
+  @doc("The end of the time range from which to upload Batch Service log file(s). Any log file containing a log message in the time range will be uploaded. This means that the operation might retrieve more logs than have been requested since the entire log file is always uploaded, but the operation should not retrieve fewer logs than have been requested. If omitted, the default is to upload all logs available after the startTime.")
+  endTime?: utcDateTime;
+
+  @doc("The reference to the user assigned identity to use to access Azure Blob Storage specified by containerUrl. The identity must have write access to the Azure Blob Storage container.")
+  identityReference?: BatchNodeIdentityReference;
+}
+
+@doc("The result of uploading Batch service log files from a specific Compute Node.")
+@access(Access.public, "java")
+model UploadBatchServiceLogsResult {
+  @doc("The virtual directory within Azure Blob Storage container to which the Batch Service log file(s) will be uploaded. The virtual directory name is part of the blob name for each log file uploaded, and it is built based poolId, nodeId and a unique identifier.")
+  virtualDirectoryName: string;
+
+  @doc("The number of log files which will be uploaded.")
+  numberOfFilesUploaded: int32;
+}
+
+@doc("The result of listing the Compute Nodes in a Pool.")
+@pagedResult
+@access(Access.public, "java")
+model BatchNodeListResult {
+  @doc("The list of Compute Nodes.")
+  @items
+  value?: BatchNode[];
+
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Property types must use camelCase"
+  @doc("The URL to get the next set of results.")
+  @nextLink
+  `odata.nextLink`?: string;
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style"
+@doc("The configuration for virtual machine extension instance view.")
+@access(Access.public, "java")
+model BatchNodeVMExtension {
+  @doc("The provisioning state of the virtual machine extension.")
+  provisioningState?: string;
+
+  @doc("The virtual machine extension.")
+  vmExtension?: VMExtension;
+
+  @doc("The vm extension instance view.")
+  instanceView?: VMExtensionInstanceView;
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Property types must use camelCase"
+@doc("The vm extension instance view.")
+@access(Access.public, "java")
+model VMExtensionInstanceView {
+  @doc("The name of the vm extension instance view.")
+  name?: string;
+
+  @doc("The resource status information.")
+  statuses?: InstanceViewStatus[];
+
+  @doc("The resource status information.")
+  subStatuses?: InstanceViewStatus[];
+}
+
+@doc("The instance view status.")
+@access(Access.public, "java")
+model InstanceViewStatus {
+  @doc("The status code.")
+  code?: string;
+
+  @doc("The localized label for the status.")
+  displayStatus?: string;
+
+  @doc("Level code.")
+  level?: StatusLevelTypes;
+
+  @doc("The detailed status message.")
+  message?: string;
+
+  @doc("The time of the status.")
+  time?: string;
```

Comment:
`utcDateTime`?

Reply:
this is how its defined in the original swagger

Reply:
changed to utdDateTime